src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -244,6 +253,37 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
 
                     return true;
                 }
+
+                static (bool includeSyntax, bool includeSemantic) ProcessDiagnosticKinds(
+                    DiagnosticAnalyzer analyzer,
+                    DiagnosticKinds diagnosticKinds,
+                    ref bool seenCompilerAnalyzer)
+                {
+                    if (diagnosticKinds == DiagnosticKinds.All)
```

Comment:
i technically don't like thsi.  this logic should fall out from below, so i'd prefer to remove it.

Reply:
The goal here is again to avoid invoking `IsCompilerAnalyzer()` API for analyzers unless the caller needs fine grained diagnostics.